templates/shared_services/sonatype-nexus/terraform/webapp.tf

```diff
@@ -0,0 +1,193 @@
+    share_name = azurerm_storage_share.nexus.name
+    mount_path = "/nexus-data"
+  }
+
+  logs {
+    application_logs {
+      file_system_level = "Information"
+    }
+
+    http_logs {
+      file_system {
+        retention_in_days = 7
+        retention_in_mb   = 100
+      }
+    }
+  }
+}
+
+resource "azurerm_private_endpoint" "nexus_private_endpoint" {
+  name                = "pe-nexus-${var.tre_id}"
+  resource_group_name = local.core_resource_group_name
+  location            = var.location
+  subnet_id           = data.azurerm_subnet.shared.id
+
+  private_service_connection {
+    private_connection_resource_id = azurerm_app_service.nexus.id
+    name                           = "psc-nexus-${var.tre_id}"
+    subresource_names              = ["sites"]
+    is_manual_connection           = false
+  }
+
+  private_dns_zone_group {
+    name                 = "privatelink.azurewebsites.net"
+    private_dns_zone_ids = [data.azurerm_private_dns_zone.azurewebsites.id]
+  }
+}
+
+resource "azurerm_app_service_virtual_network_swift_connection" "nexus-integrated-vnet" {
+  app_service_id = azurerm_app_service.nexus.id
+  subnet_id      = data.azurerm_subnet.web_app.id
+}
+
+resource "azurerm_monitor_diagnostic_setting" "nexus" {
+  name                       = "diag-${var.tre_id}"
+  target_resource_id         = azurerm_app_service.nexus.id
+  log_analytics_workspace_id = data.azurerm_log_analytics_workspace.tre.id
+
+  log {
+    category = "AppServiceHTTPLogs"
+    enabled  = true
+
+    retention_policy {
+      days    = 1
+      enabled = false
+    }
+  }
+
+  log {
+    category = "AppServiceConsoleLogs"
+    enabled  = true
+
+    retention_policy {
+      days    = 1
+      enabled = false
+    }
+  }
+
+  log {
+    category = "AppServiceAppLogs"
+    enabled  = true
+
+    retention_policy {
+      days    = 1
+      enabled = false
+    }
+  }
+
+  log {
+    category = "AppServiceFileAuditLogs"
+    enabled  = true
+
+    retention_policy {
+      days    = 1
+      enabled = false
+    }
+  }
+
+  log {
+    category = "AppServiceAuditLogs"
+    enabled  = true
+
+    retention_policy {
+      days    = 1
+      enabled = false
+    }
+  }
+
+  log {
+    category = "AppServiceIPSecAuditLogs"
+    enabled  = true
+
+    retention_policy {
+      days    = 1
+      enabled = false
+    }
+  }
+
+  log {
+    category = "AppServicePlatformLogs"
+    enabled  = true
+
+    retention_policy {
+      days    = 1
+      enabled = false
+    }
+  }
+
+  log {
+    category = "AppServiceAntivirusScanAuditLogs"
+    enabled  = true
+
+    retention_policy {
+      days    = 1
+      enabled = false
+    }
+  }
+
+  metric {
+    category = "AllMetrics"
+    enabled  = true
+
+    retention_policy {
+      enabled = false
+    }
+  }
+}
+
+resource "azurerm_storage_share" "nexus" {
+  name = "nexus-data"
+  storage_account_name = data.azurerm_storage_account.nexus.name
+  quota                = 1024 # 1TB 
```

Comment:
Can see this being hit if we actually mirror anything rather than proxy. Might be worth adding an issue to track/make a variable so can easily be increased.

Reply:
Added a var